### PR TITLE
WebID-OIDC authentication spec proposal

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
First draft of spec proposal, outline of motivation, summary of workflow,
detailed workflow example.

To address issue #22.

### CODE REVIEW

**Comment by deiu** [comment]
👍

File: proposals/webid-oidc/README.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+* [Detailed Sign In Workflow Example](#detailed-sign-in-workflow-example)
+
+## Introduction
+
+WebID-OIDC is an authentication delegation protocol (as well as a toolkit of
+useful auth-related verification techniques) suitable for WebID-based
+decentralized systems such as [Solid](https://github.com/solid/solid), as well
+as most LDP-based systems.
+
+The end result of any WebID authentication workflow is a verified WebID
```
Comment (dan-f) [inline proposals/webid-oidc/README.md]: I think the way you said it is great. Verifying that an agent owns a WebID. Without the agent part "verified WebID URI" sounds like the use case for a certificate authority.

Comment (dmitrizagidulin) [inline proposals/webid-oidc/README.md]: Thanks! How's this latest wording?

**Comment by dan-f** [comment]
Sorry, did that thing again where I forgot to "submit" my review.

**Comment by dmitrizagidulin** [comment]
@dan-f Oh, ack, I totally missed the notification for these comments! Sorry bout that.

File: proposals/webid-oidc/README.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+proposals including Bitcoin, and other client-side key management based systems.
+
+If at all possible, we were looking for protocols that used standards and tech
+stacks that were familiar and comprehensible to app developers (which ruled out
+the XML-based and enterprise-heavy SAML, as well as earlier incarnations of
+OpenID).
+
+Other properties that we were looking for in a protocol included:
+
+* Support for the full range of app and agent types, including browser side
```
Comment (dmitrizagidulin) [inline proposals/webid-oidc/README.md]: Support means "able to be used by" a full range of app types.

File: proposals/webid-oidc/README.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+side Javascript apps, for example) do not have access to private key storage
+facilities or keychain APIs. (And even the upcoming WebAuthn spec requires the
+user to *register* an account for each domain -- precisely the situation that
+most decentralized social app projects like Solid are trying to fix.) This
+ruled out proposals such as HTTP Signatures, most "Identity on the Blockchain"
+proposals including Bitcoin, and other client-side key management based systems.
+
+If at all possible, we were looking for protocols that used standards and tech
+stacks that were familiar and comprehensible to app developers (which ruled out
+the XML-based and enterprise-heavy SAML, as well as earlier incarnations of
```
Comment (dmitrizagidulin) [inline proposals/webid-oidc/README.md]: Sure, that's an ambiguous (and subjective) adjective, I can take that out.

**Pull request merged**